Kitodo/src/main/java/org/kitodo/production/helper/MetadataComparison.java

```diff
@@ -0,0 +1,275 @@
+    }
+
+    private boolean canAdd() {
+        return metadataView.getMinOccurs() <= oldValues.size() + newValues.size()
+                && oldValues.size() + newValues.size() <= metadataView.getMaxOccurs();
+    }
+
+    private boolean canReplace() {
+        return newValues.size() >= metadataView.getMinOccurs()
+                && newValues.size() <= metadataView.getMaxOccurs();
+    }
+
+    /**
+     * Convert given set of metadata to list of strings and returns it.
+     *
+     * @param metadataSet set of metadata to be converted to strings
+     *
+     * @return list of string representing given metadata
+     */
+    public List<String> getValuesAsStrings(HashSet<Metadata> metadataSet) {
+        return metadataSet.stream()
+                .map(DataEditorService::metadataToString)
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Convert given metadata to string and return it.
+     *
+     * @param metadata metadata to be converted to string
+     *
+     * @return string conversion of given metadata
+     */
+    public String getValueAsString(Metadata metadata) {
+        return DataEditorService.metadataToString(metadata);
+    }
+
+    /**
+     * Return list of given metadata groups metadata, sorted alphabetically by their string representation.
+     *
+     * @param metadataGroup MetadataGroup whose metadata is sorted and returned
+     *
+     * @return list of metadata sorted by string representation
+     */
+    public List<Metadata> getMetadataSorted(MetadataGroup metadataGroup) {
+        return metadataGroup.getMetadata().stream().sorted(Comparator.comparing(Metadata::getKey))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * List of old metadata groups sorted by their 'groupDisplayLabel'.
+     *
+     * @return old metadata groups sorted by their 'groupDisplayLabel'.
+     */
+    public List<Metadata> getOldMetadataGroupsSorted(Ruleset ruleset) {
+        if (isMetadataGroup) {
+            try {
+                return RulesetService.getGroupsSortedByGroupDisplayLabel(oldValues, ruleset);
+            } catch (IOException e) {
+                Helper.setErrorMessage(e);
+                return new ArrayList<>();
+            }
+        } else {
+            return getValuesSorted(oldValues);
+        }
+    }
+
+    /**
+     * List of new metadata groups sorted by their 'groupDisplayLabel'.
+     *
+     * @return new metadata groups sorted by their 'groupDisplayLabel'.
+     */
+    public List<Metadata> getNewMetadataGroupsSorted(Ruleset ruleset) {
+        if (isMetadataGroup) {
+            try {
+                return RulesetService.getGroupsSortedByGroupDisplayLabel(newValues, ruleset);
+            } catch (IOException e) {
+                Helper.setErrorMessage(e);
+                return new ArrayList<>();
+            }
+        } else {
+            return getValuesSorted(newValues);
+        }
+    }
```

Comment:
This could be refactored to a single private method (getMetadataGroupsSorted(Ruleset ruleset, List<Metadata> values)) that accepts the values as a parameter,

Reply:
Done.